### PR TITLE
PR #48 (theme-fixes)

### PR DESCRIPTION
**Pull request by rgadon107**
## PR Summary

This PR includes several miscellaneous fixes to the `developers` child theme. 

1) Fixes a call to `wp_get_theme()->get( 'Version' )` in the child theme's `functions.php` file that threw an error in `Whoops!`

2) Renamed a registered widget area from `donate_call_to_action` to `footer_widget_area`. 

3) Removed a call to `site-footer.php`, and the view file itself that hard-coded the client's contact info into the footer. The info is now rendered in a footer widget accessible from the WP admin. This gives the client more control to change the footer in the future. 

4) Updates the value of `$rem-baseline` set in `/developers/dev/node_modules/rem-sass/_rem.scss`. The values set in the child theme are found in `typography.scss` and `styles.scss` with notes. 

5) Rebuilt the child theme styles after resetting the value of `$rem-baseline`. 

6) Updates the `.gitignore` file to remove the `/bin` and `/themes/developers/node_modules` directories.

### CODE REVIEW

**Comment by hellofromtonya** [comment]
@rgadon107 You want to override `sass-rem`'s `$rem-baseline` _within your theme's Sass files_ and _**not** within the node module_ itself.

Tip: Never ever directly modify code within a node module. Why? When an update happens, your changes will be overwritten. Rather, only change your code.

In this case, `sass-rem` gives you a variable for your theme to override the node module's behavior. The module gives you [instructions for changing the baseline, found here](https://www.npmjs.com/package/sass-rem#changing-baseline). I'd suggest the following to the `developers/assets/sass/style.scss` partial:

```
@import '../../node_modules/sass-rem/rem';

$rem-baseline: 10px;
$rem-fallback: true;
```

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Directories
 /ewww
+/bin
```
Comment (hellofromtonya) [inline .gitignore]: @rgadon107 We actually want to keep `/bin` in the repo. We'll strip it out when it comes time to do a formal release.

File: themes/developers/dev/assets/sass/utilities/variables/_typography.scss [hunk 2]
```diff
@@ -4,6 +4,7 @@ $font-family--lato: 'Lato', sans-serif;
 
 // Base Typography
 $html--font-size: 62.5%;
+$rem-baseline: 10px;
```
Comment (hellofromtonya) [inline themes/developers/dev/assets/sass/utilities/variables/_typography.scss]: @rgadon107 I'd suggest:

1. moving the rem fallback parameter here too, ie to keep all of the REM configuration parameters together
2. Moving both of them up to the top of this Sass partial and adding a comment for them.

File: .gitignore (hunk 1 above)
Comment (rgadon107) [reply .gitignore]: @hellofromtonya Oh, I didn't realize that. Will remove.

**Comment by rgadon107** [comment]
> @rgadon107 You want to override `sass-rem`'s `$rem-baseline` _within your theme's Sass files_ and _**not** within the node module_ itself.
> 
> Tip: Never ever directly modify code within a node module. Why? When an update happens, your changes will be overwritten. Rather, only change your code.
> 
> In this case, `sass-rem` gives you a variable for your theme to override the node module's behavior. The module gives you [instructions for changing the baseline, found here](https://www.npmjs.com/package/sass-rem#changing-baseline). I'd suggest the following to the `developers/assets/sass/style.scss` partial:
> 
> ```
> @import '../../node_modules/sass-rem/rem';
> 
> $rem-baseline: 10px;
> $rem-fallback: true;
> ```
@hellofromtonya Sorry I didn't look at this sooner. I didn't realize that `$rem-baseline` should be set in the `style.scss partial file as well as the `_typography.scss` partial file. Will make the adjustment. 

>Tip: Never ever directly modify code within a node module. Why? When an update happens, your changes will be overwritten. Rather, only change your code.

I thought that was the case. I didn't fully understand your instructions from yesterday afternoon until I reread them this morning. I'm in full agreement with you on this.

File: .gitignore [hunk 3]
```diff
@@ -26,7 +28,7 @@
 /plugins/wp-migrate-db-pro
 /plugins/wp-migrate-db-pro-media-files
 /themes/Beans
-/themes/developers/node_modules/
+/themes/developers/dev/node_modules/
```
Comment (hellofromtonya) [inline .gitignore]: @rgadon107 This is unnecessary. The theme has a `.gitignore` that already excludes the node modules. So you can just delete this line. :)

**Comment by hellofromtonya** [comment]
>Sorry I didn't look at this sooner. I didn't realize that $rem-baseline should be set in the style.scss partial file as well as the _typography.scss` partial file. Will make the adjustment.

@rgadon107  You only need to define it in one place. You can do it in the `_typography.scss` or `style.scss`. It might make more sense in the `_typography.scss` partial, as that's where you are configuring all of the font sizes.

File: themes/developers/dev/assets/sass/utilities/variables/_typography.scss (hunk 2 above)
Comment (rgadon107) [reply themes/developers/dev/assets/sass/utilities/variables/_typography.scss]: @hellofromtonya Done.

File: .gitignore (hunk 3 above)
Comment (rgadon107) [reply .gitignore]: @hellofromtonya I removed the reference & updated the .gitignore for the child theme.

**Pull request merged**